ctrtool/syscalls.c

```diff
@@ -0,0 +1,165 @@
+	NULL,                                // 42
+	NULL,                                // 43
+	NULL,                                // 44
+	NULL,                                // 45
+	NULL,                                // 46
+	"CreatePort",                        // 47
+	"CreateSessionToPort",               // 48
+	"CreateSession",                     // 49
+	"AcceptSession",                     // 4A
+	"ReplyAndReceive1",                  // 4B
+	"ReplyAndReceive2",                  // 4C
+	"ReplyAndReceive3",                  // 4D
+	"ReplyAndReceive4",                  // 4E
+	"ReplyAndReceive",                   // 4F
+	"BindInterrupt",                     // 50
+	"UnbindInterrupt",                   // 51
+	"InvalidateProcessDataCache",        // 52
+	"StoreProcessDataCache",             // 53
+	"FlushProcessDataCache",             // 54
+	"StartInterProcessDma",              // 55
+	"StopDma",                           // 56
+	"GetDmaState",                       // 57
+	"RestartDma",                        // 58
+	NULL,                                // 59
+	NULL,                                // 5A
+	NULL,                                // 5B
+	NULL,                                // 5C
+	NULL,                                // 5D
+	NULL,                                // 5E
+	NULL,                                // 5F
+	"DebugActiveProcess",                // 60
+	"BreakDebugProcess",                 // 61
+	"TerminateDebugProcess",             // 62
+	"GetProcessDebugEvent",              // 63
+	"ContinueDebugEvent",                // 64
+	"GetProcessList",                    // 65
+	"GetThreadList",                     // 66
+	"GetDebugThreadContext",             // 67
+	"SetDebugThreadContext",             // 68
+	"QueryDebugProcessMemory",           // 69
+	"ReadProcessMemory",                 // 6A
+	"WriteProcessMemory",                // 6B
+	"SetHardwareBreakPoint",             // 6C
+	"GetDebugThreadParam",               // 6D
+	NULL,                                // 6E
+	NULL,                                // 6F
+	"ControlProcessMemory",              // 70
+	"MapProcessMemory",                  // 71
+	"UnmapProcessMemory",                // 72
+	"CreateCodeSet",                     // 73
+	NULL,                                // 74
+	"CreateProcess",                     // 75
+	"TerminateProcess",                  // 76
+	"SetProcessResourceLimits",          // 77
+	"CreateResourceLimit",               // 78
+	"SetResourceLimitValues",            // 79
+	"AddCodeSegment",                    // 7A
+	"Backdoor",                          // 7B
+	"KernelSetState",                    // 7C
+	"QueryProcessMemory",                // 7D
+	NULL,                                // 7E
+	NULL,                                // 7F
+};
+
+
+void syscall_get_name(char *output, size_t size, unsigned int call_num)
+{
+	typedef char StaticAssert[sizeof(syscall_list) / sizeof(syscall_list[0]) == NUM_SYSCALLS ? 1 : -1];
```

Comment:
It's like the C11 keyword ```_Static_assert``` or C++11 keyword ```static_assert```--if the expression is false at compile time, a compiler error will occur, failing to build.  However, Visual Studio doesn't have ```_Static_assert``` in C, only ```static_assert``` in C++.

If the expression is true, that line declares a local typedef named ```StaticAssert``` of type ```char[1]```, which has no effect.  If the expression is false, that line tries to typedef ```StaticAssert``` as ```char[-1]```, which is ill-formed in C (and C++).

OK, the ```fseeko64``` is now ```#ifdef _MSC_VER``` instead of _WIN32.